app/src/sharedTest/java/org/oppia/android/app/devoptions/featureflags/FeatureFlagsFragmentTest.kt

```diff
@@ -1072,6 +1069,187 @@ class FeatureFlagsFragmentTest {
     }
   }
 
+  @Test
+  fun testFeatureFlagsFragment_navigateBackWithFlagModified_clickDiscard_discardsChanges() {
+    setUpTestApplicationComponent()
+    val initialValue = getEphemeralFeatureFlags()[0].currentValue
+
+    launch(FeatureFlagsTestActivity::class.java).use {
+      testCoroutineDispatchers.runCurrent()
+
+      scrollToPosition(0)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.feature_flags_recycler_view,
+          position = 0,
+          targetViewId = R.id.feature_flag_switch
+        )
+      ).perform(click())
+
+      pressBack()
+      testCoroutineDispatchers.runCurrent()
+
+      onView(withText(R.string.pending_changes_dialog_discard_button_text))
+        .inRoot(isDialog())
+        .check(matches(isDisplayed()))
+        .perform(click())
+      testCoroutineDispatchers.runCurrent()
+    }
+
+    launch(FeatureFlagsTestActivity::class.java).use {
+      testCoroutineDispatchers.runCurrent()
+
+      scrollToPosition(0)
+      verifyFeatureFlagSwitchState(
+        position = 0,
+        expectedState = initialValue
+      )
+    }
+  }
+
+  @Test
+  fun testFeatureFlagsFragment_saveChangesOnBackNavigation_showRestartDialogExitApp() {
```

Comment:
This sounds a bit awkward, let' try to rephrase it to be more descriptive of the test behaviour.
```suggestion
  fun testFeatureFlagsFragment_pressBack_saveChanges_clickRestartInDialog_exitsApp() {
```

Reply:
updated